### ISSUE TITLE
Revise option output_function

### ISSUE THREAD

**Issue by schillic** [JuliaReach/Reachability.jl#468]
Internally, we create the option `:output_function`. It can either be `nothing` or a matrix, say, `A`. In the latter case, before we store the reachset `X`, we apply a box approximation to the set `A * X`.
I would rename the option (nobody would expect that behavior from the name `output_function` :smile:) and refactor it out of the actual reachability code. This is a typical post-processing task.

**Comment by mforets** [comment]
Yes, i agree to refactor that code out of the reachability code. We can use a [system with output](https://juliareach.github.io/MathematicalSystems.jl/latest/lib/types.html#Systems-with-output-1) to substitute `:output_function` and do the post-processing.

**Comment by schillic** [comment]
I changed my mind here. This is probably the only way to support lazy flowpipes with lazy inputs. So it makes sense to keep it. However, it currently crashes during projection, so this needs to be revised. The option should probably be renamed and become an input option.
